src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/SyntheticTelemetryInitializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.TelemetryInitializers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Sockets;
+    using Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.AspNet.Http;
+    using Microsoft.AspNet.Http.Features;
+
+    /// <summary>
+    /// This will allow to mark synthetic traffic from availability tests
+    /// </summary>
+    public class SyntheticTelemetryInitializer : TelemetryInitializerBase
+    {
+        private const string HeaderNameDefault = "GsmSyntheticTestRunId";
+        private const string SyntheticTestRunId = "SyntheticTest-RunId";
+        private const string SyntheticTestLocation = "SyntheticTest-Location";
+
+        private const string SyntheticSourceHeaderValue = "Application Insights Availability Monitoring";
+
+        public SyntheticTelemetryInitializer(IHttpContextAccessor httpContextAccessor)
+             : base(httpContextAccessor)
+        {
+
+        }
+
+        protected override void OnInitializeTelemetry(HttpContext platformContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
+        {
+            if (string.IsNullOrEmpty(telemetry.Context.Operation.SyntheticSource))
+            {
+                var headerValue = platformContext.Request.Headers[HeaderNameDefault];
+                if (!string.IsNullOrEmpty(headerValue))
+                {
+                    telemetry.Context.Operation.SyntheticSource = SyntheticSourceHeaderValue;
+
+                    var locationHeader = platformContext.Request.Headers[SyntheticTestLocation];
```

Comment:
Can we set values only when all headers are present like we do in Web? https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/master/Src/Web/Web.Shared.Net/WebTestTelemetryInitializer.cs#L36